### PR TITLE
fix(errors): better error message for hot/cold repo in check

### PR DESCRIPTION
**Pull request by simonsan**
Fixes https://github.com/rustic-rs/rustic/issues/1229

### CODE REVIEW

File: crates/core/src/commands/check.rs [hunk 1]
```diff
@@ -447,7 +447,9 @@ fn check_packs_list(be: &impl ReadBackend, mut packs: HashMap<PackId, u32>) -> R
         .map_err(RusticErrorKind::Backend)?
     {
         match packs.remove(&PackId::from(id)) {
-            None => warn!("pack {id} not referenced in index. Can be a parallel backup job. To repair: 'rustic repair index'."),
+            None => {
+                warn!("(hot) pack {id} not referenced as tree pack in index, but as data pack.");
```
Comment (aawsome) [inline crates/core/src/commands/check.rs:451]: this is currently used for both hot packs and for normal (maybe cold) packs. So changing this this way makes it correct for hot packs, but wrong in the standard case.

Comment (simonsan) [reply crates/core/src/commands/check.rs:451]: Ah! How would we differentiate in `None` between hot and cold packs, then I can implement it.

Comment (aawsome) [reply crates/core/src/commands/check.rs]: I copied the check_pack function and now handle checking of hot packs differently from normal packs. Also added a test about data packs contained in the hot repo.

Comment (simonsan) [reply crates/core/src/commands/check.rs]: Is duplicating the function and code the only good way to handle it, currently? Do you think we will implement different handling (checking) for hot and cold files in the future, so we could reuse this?

Comment (aawsome) [reply crates/core/src/commands/check.rs]: It is already different. In the hot pack check we now additionally check against the total pack list to give another error in the specific situation the user had...

File: crates/core/src/commands/check.rs [hunk 2]
```diff
@@ -456,6 +456,46 @@ fn check_packs_list(be: &impl ReadBackend, mut packs: HashMap<PackId, u32>) -> R
     Ok(())
 }
 
+///Checks if all packs in the backend are also in the index
```
Comment (simonsan) [inline crates/core/src/commands/check.rs:459]:
```suggestion
/// Checks if all packs in the backend are also in the index
```

File: crates/core/src/commands/check.rs [hunk 3]
```diff
@@ -426,7 +426,7 @@ fn check_packs(
 }
 
 // TODO: Add documentation
-/// Checks if all packs in the backend are also in the index
+///Checks if all packs in the backend are also in the index
```
Comment (simonsan) [inline crates/core/src/commands/check.rs:429]:
```suggestion
/// Checks if all packs in the backend are also in the index
```

**Pull request merged**